test/framework/machinedeployment_helpers.go

```diff
@@ -76,6 +77,20 @@ func GetMachineDeploymentsByCluster(ctx context.Context, input GetMachineDeploym
 	return deployments
 }
 
+// GetMachinePoolsByCluster returns the MachinePools objects for a cluster.
+// Important! this method relies on labels that are created by the CAPI controllers during the first reconciliation, so
+// it is necessary to ensure this is already happened before calling it.
+func GetMachinePoolsByCluster(ctx context.Context, input GetResourcesByClusterInput) []*clusterv1exp.MachinePool {
```

Comment:
Should this goes in a separated file named `machinedpool_helpers.go` (same for other new types/func below)

Reply:
As suggested, I moved the `MachinePool` functionality to a new file.